### PR TITLE
[USH-2037] extension sync bug

### PR DESCRIPTION
**Pull request by snopoke**
## Technical Summary
Jira: https://dimagi-dev.atlassian.net/browse/USH-2307

The above ticket describes an issue where a case should be 'on the device' but it is not.

Starting point:

```
host <--ext-- claim (owned)
     <--ext-- client
```

* Removing the extension index from 'client' should remove the case from the phone
* Closing the 'host' case should remove the 'claim' case from the phone

Both of these should result in the synclog being updated to remove the case ID so that future incremental syncs are correct.

This PR fixes both of these issues:

1. Closing extension cases after closing a host case wasn't updating the synclogs
  * I think this is causing this error: https://sentry.io/share/issue/ebca3f40b3fb475e866fe264b892aac1/
2. Removing and index from an extension case prevents it from being purged from the synclog

## Feature Flag
Extension sync

### Safety story
The sync code is well tested and the rules are well understood. There is good test coverage for other scenarios.

### Automated test coverage
Tests added for these edge cases, other tests already exist for other scenarios.

### QA Plan
None

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Comment by calellowitz** [comment]
Do we need to update https://github.com/dimagi/commcare-hq/blob/master/corehq/ex-submodules/casexml/apps/phone/tests/data/case_relationship_tests.json for the new behavior?

**Comment by snopoke** [comment]
> Do we need to update https://github.com/dimagi/commcare-hq/blob/master/corehq/ex-submodules/casexml/apps/phone/tests/data/case_relationship_tests.json for the new behavior?

No. There is no change to the clean sync behaviour. This was a bug with incremental sync.

**Review by esoergel — APPROVED** [review]
Nice tests!  Though I was thrown by `ferrel` until I grepped the file

**Review by calellowitz — APPROVED** [review]

File: corehq/ex-submodules/casexml/apps/phone/data_providers/case/livequery.py [hunk 1]
```diff
@@ -82,7 +82,11 @@ def do_livequery(timing_context, restore_state, response, async_task=None):
                 sync_ids = discard_already_synced_cases(live_ids, restore_state)
         else:
             sync_ids = live_ids
+
+        dependent_ids = live_ids - set(owned_ids)
+        debug('updating synclog: live=%r dependent=%r', live_ids, dependent_ids)
         restore_state.current_sync_log.case_ids_on_phone = live_ids
+        restore_state.current_sync_log.dependent_case_ids_on_phone = dependent_ids
```
Comment (millerdev) [inline corehq/ex-submodules/casexml/apps/phone/data_providers/case/livequery.py:89]: Was this part of the bug? I don't see any other code in this PR that references this field. Trying to figure out if it was an oversight in the original writing of livequery?

Comment (snopoke) [reply corehq/ex-submodules/casexml/apps/phone/data_providers/case/livequery.py:89]: This was the bug. This field is referenced in the synclog model and is used when updating the synclog after a form submission. Without this field being populated all cases on the phone are assumed to be 'primary' cases and won't be purged from the synclog. 

https://github.com/dimagi/commcare-hq/blob/1992a5737194eef7c7546057b7fb08e4834d44be/corehq/ex-submodules/casexml/apps/phone/models.py#L636-L638

As you are aware, purging cases from the synclog is essential to the incremental sync algorithm since CC won't send cases to the phone that it thinks are already there.
